src/prototype_room_utils.js

```diff
@@ -133,24 +133,67 @@ Room.prototype.checkRoleToSpawn = function(role, amount, targetId, targetRoom) {
   this.memory.queue.push(creepMemory);
 };
 
-Room.prototype.getPartConfig = function(energy, parts) {
-  var sum = 0;
-  var i = 0;
-  var partConfig = [];
-  while (sum < energy && partConfig.length < 50) {
-    var part = parts[i % parts.length];
-    if (sum + BODYPART_COST[part] <= energy) {
-      partConfig.push(part);
-      sum += BODYPART_COST[part];
-      i += 1;
-    } else {
-      break;
-    }
-  }
-  return partConfig;
+Room.prototype.getPartConfig = function(datas) {
+
+  let layout = datas.layout;
+  let amount = datas.amount;
+  let minEnergyStored = datas.minEnergyStored;
+  let maxEnergyUsed = datas.maxEnergyUsed;
+  let bonusParts = datas.bonusParts;
```

Comment:
would you mind to add a datas.prefixParts? 
see: https://github.com/TooAngel/screeps/pull/54/files#diff-d82ead92982e33be3e93b47c11bcb784L20
so that 
```js
{
// ....
prefixParts: [MOVE, CARRY, WORK],
layout: [MOVE, WORK, WORK]
// ....
}
```
becomes like [MOVE, CARRY, WORK].concat(_.repeat( [MOVE, WORK, WORK], 6))